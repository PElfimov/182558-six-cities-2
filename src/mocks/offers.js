const offers = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 120,
    name: `Beautiful & luxurious apartment at great location`,
    rating: 55,
    type: `Apartment`,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: false,
    cost: 132,
    name: `Wood and stone place`,
    rating: 80,
    type: `Apartment`,
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    id: 3,
    city: {
      name: `Amsterdam`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 120,
    name: `Canal View Prinsengracht`,
    rating: 55,
    type: `Apartment`,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 4,
    city: {
      name: `Amsterdam`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 80,
    name: `Nice, cozy, warm big bed apartment`,
    rating: 100,
    type: `Apartment`,
    coordinates: [52.3809553943508, 4.939309666406198]
  },

  {
    id: 5,
    city: {
      name: `Biysk`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 80,
    name: `Nice, cozy, warm big bed apartment`,
    rating: 100,
    type: `Apartment`,
    coordinates: [52.3809553943508, 4.939309666406195]
  },
  {
    id: 6,
    city: {
      name: `Barnaul`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 80,
    name: `Nice, cozy, warm big bed apartment`,
    rating: 100,
    type: `Apartment`,
    coordinates: [52.3809553943508, 4.939309666406195]
  },
  {
    id: 7,
    city: {
      name: `Moscow`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 80,
    name: `Nice, cozy, warm big bed apartment`,
    rating: 100,
    type: `Apartment`,
    coordinates: [52.3809553943508, 4.939309666406195]
  },
  {
    id: 8,
    city: {
      name: `St Petersburg`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 120,
    name: `Canal View Prinsengracht`,
    rating: 55,
    type: `Apartment`,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 8,
    city: {
      name: `Hamburg`,
      coordinates: [52.38333, 4.9]
    },
    isPremium: true,
    cost: 120,
    name: `Canal View Prinsengracht`,
    rating: 55,
    type: `Apartment`,
    coordinates: [52.3909553943508, 4.929309666406198]
  },

];

export default offers;
