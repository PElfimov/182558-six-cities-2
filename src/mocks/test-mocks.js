const offers = [
  {
    id: 1,
    isPremium: true,
    cost: 120,
    name: `Beautiful & luxurious apartment at great location`,
    rating: 55,
    type: `Apartment`,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    id: 2,
    isPremium: false,
    cost: 132,
    name: `Wood and stone place`,
    rating: 80,
    type: `Apartment`,
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    id: 3,
    isPremium: true,
    cost: 120,
    name: `Canal View Prinsengracht`,
    rating: 55,
    type: `Apartment`,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 4,
    isPremium: true,
    cost: 80,
    name: `Nice, cozy, warm big bed apartment`,
    rating: 100,
    type: `Apartment`,
    coordinates: [52.3809553943508, 4.939309666406198]
  }];
const activeCity = [52.38333, 4.9];

const getCitiesListFromOffers = jest.fn();
const getFilteredOffers = jest.fn();
const toString = jest.fn();

export {offers, activeCity, getCitiesListFromOffers, getFilteredOffers, toString};
