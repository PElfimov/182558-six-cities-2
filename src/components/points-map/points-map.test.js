import React from "react";
import renderer from "react-test-renderer";
import {offers, activeCity} from "../../mocks/testMocks";
import PointsMap from './points-map';

jest.mock(`./points-map`, () => jest.fn().mockReturnValue(null));

it(`Maps correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<PointsMap offers={offers} city={activeCity} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
