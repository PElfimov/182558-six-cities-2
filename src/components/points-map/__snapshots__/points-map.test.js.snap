// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps correctly renders after relaunch 1`] = `null`;
